controller/array_action/array_mediator_svc.py

```diff
@@ -764,6 +764,65 @@ def get_host_by_host_identifiers(self, initiators):
             raise array_errors.MultipleHostsFoundError(initiators, fc_host)
         return host_name, list(connectivity_types)
 
+    def _get_host_names_by_wwpn(self, host_wwpn):
+        fabrics = self._lsfabric(wwpn=host_wwpn).as_list
+        return set(fabric.name for fabric in fabrics)
+
+    def _lsnvmefabric(self, host_nqn):
+        try:
+            return self.client.svcinfo.lsnvmefabric(remotenqn=host_nqn).as_list
+        except(svc_errors.CommandExecutionError, CLIFailureError) as ex:
+            logger.error("Failed to get nvme fabrics. Reason "
+                         "is: {0}".format(ex))
+            raise ex
+
+    def _get_host_names_by_nqn(self, nqn):
+        nvme_fabrics = self._lsnvmefabric(nqn)
+        return set(nvme_fabric.object_name for nvme_fabric in nvme_fabrics)
+
+    def _lshostiplogin(self, iqn):
+        try:
+            return self.client.svcinfo.lshostiplogin(object_id=iqn).as_single_element
+        except(svc_errors.CommandExecutionError, CLIFailureError) as ex:
+            if SPECIFIED_OBJ_NOT_EXIST in ex.my_message:
+                return None
+            logger.error("Failed to get iscsi host. Reason "
+                         "is: {0}".format(ex))
+            raise ex
+
+    def _get_host_name_by_iqn(self, iqn):
+        iscsi_login = self._lshostiplogin(iqn)
+        if iscsi_login:
+            return iscsi_login.host_name
+        return None
+
+    def get_host_by_host_identifiers(self, initiators):
+        logger.debug("Getting host name for initiators : {0}".format(initiators))
+        host_names = set()
+        connectivity_types = set()
+        for connectivity_type, initiator in initiators:
+            if connectivity_type == config.NVME_OVER_FC_CONNECTIVITY_TYPE:
+                nvme_host_names = self._get_host_names_by_nqn(initiator)
+                if nvme_host_names:
+                    host_names.update(nvme_host_names)
```

Comment:
why are you adding it to a set if it's already returned as a set?

Reply:
`host_names` is a set of all the host names found with relevant initiators defined (for every worker connectivity type).
what's wrong with adding a set to a set? what are you suggesting to do?

Reply:
never mind my bad I missed